docs/dev/data-identity.rst

```diff
@@ -0,0 +1,250 @@
+.. _data_identity:
+
+###########################
+DFHack Data Identity System
+###########################
+
+This article is an attempt to describe DFHack's data identity system.
+DFHack internally has a collection of C++ classes that provide metedata about the data in DF itself as well as data used
+by various components of DFHack. This metadata is used primarily to enable the Lua scripting system to access data
+held by Dwarf Fortress in a transparent manner, but is also used for several other purposes within DFHack.
+
+The base class of the identity system is the class ``type_identity``, defined in :source:`DataDefs.h <library/include/DataDefs.h>`. A ``type_identity`` object
+provides information about one *type* of data object, in either Dwarf Fortress or DFHack, that can be manipulated as a discrete entity in Lua.
+With one specific exception (``global_identity``), there is a one-to-one relationship between C++ types and ``type_identity`` objects.
+In Lua, objects that are being managed via the data identity system are represented as a Lua userdata object. The userdata object
+contains both a pointer to the C++ object itself and a pointer to a ``type_identity`` object that describes the data pointed
+by that pointer. Note that the userdata object does not own the objects pointed to by these pointers, and the Lua engine is
+never responsible for managing their lifetimes.
+
+``type_identity`` defines the following public methods:
+
+- ``byte_size``: returns the size, in bytes, of the object held
+
```

Comment:
you don't need newlines between lines in a list, just before the first and after the last